packages/models/files.js

```diff
@@ -7,6 +7,12 @@ const Files = new FilesCollection({
     debug: true,
     collectionName: 'files',
     protected(fileObj) {
+        var session = this.params.query.xmtok;
+        var userId  = (Meteor.server.sessions[session] && Meteor.server.sessions[session].userId) ? Meteor.server.sessions[session].userId : null;
```

Comment:
Hard to find any proper documentation, it seems that there is no official API available. From what I can tell the token validation is mostly handled [in accounts_server.js](https://github.com/meteor/meteor/blob/90cb62598e68b367930ed1ffe0e69bd585c1ed91/packages/accounts-base/accounts_server.js#L970-L1063) , simply hash the token in prior and fetch it like 

```
var doc = Meteor.users.findOne({"services.resume.loginTokens.hashedToken": hashedToken}, {fields: {_id: 1}})
var userId = doc && doc._id;
```

But this solution is whacky since token can expire and you usually do not want to duplicate security related code. Could not find anything better than that for the moment.